docs/manpage.rst

```diff
@@ -386,6 +386,61 @@ Options controlling ReFrame execution
    .. versionchanged:: 3.6.1
       Multiple report files are now accepted.
 
+.. option:: -S, --setvar=[TEST.]VAR=VAL
+
+   Set variable ``VAR`` in all tests or optionally only in test ``TEST`` to ``VAL``.
+
+   Multiple variables can be set at the same time by passing this option multiple times.
+   This option *cannot* change arbitrary test attributes, but only test variables declared with the :attr:`~reframe.core.pipeline.RegressionMixin.variable` built-in.
+   If an attempt is made to change an inexistent variable or a test parameter, a warning will be issued.
+
+   ReFrame will try to convert ``VAL`` to the type of the variable.
+   If it does not succeed, a warning will be issued and the variable will not be set.
+   ``VAL`` can take the special value ``@none`` to denote that the variable must be set to :obj:`None`.
+
+   Sequence and mapping types can also be set from the command line by using the following syntax:
+
+   - Sequence types: ``-S seqvar=1,2,3,4``
+   - Mapping types: ``-S mapvar=a:1,b:2,c:3``
+
+   Conversions to arbitrary objects are also supported.
+   See :class:`~reframe.utility.typecheck.ConvertibleType` for more details.
+
+
+   The optional ``TEST.`` prefix refers to the test class name (*not* the test name).
+   If the test name is not explicitly set through the :attr:`~reframe.core.pipeline.RegressionTest.name` attribute or if the test is not parametrised, then the class name is also the test name.
```

Comment:
I am trying to see it from a user perspective that may not be so familiar with ReFrame tests. In such cases, it can be confusing for them the "test class name" vs the "test name," given also that tests are listed by name. I could make it the sentence a bit less specific here by saying sth like "generally, the test name is the same as the test class name" and refer to the `name` attribute, which btw needs to be expanded, because we don't mention anything about the default name.

Reply:
My point is that there is no need for a reference to the `name` attribute at all. The first sentence says that `TEST` is the  class name, not the test name. And the second one explains the default values of the test `name` attribute, even though the previous sentence said that this is attribute has nothing to do with this command line option.

To remove the ambiguity, I'd simply change this to  _The optional ``TEST.`` prefix refers to the test class._

Reply:
Ok, I removed completely the sentence with the reference to the `name` attribute.